File: backend/myalkansya/src/main/java/edu/cit/myalkansya/dto/CurrencyConversionResponse.java
package edu.cit.myalkansya.dto;

import java.time.LocalDateTime;

public class CurrencyConversionResponse {
    private String fromCurrency;
    private String toCurrency;
    private double amount;
    private double convertedAmount;
    private double exchangeRate;
    private LocalDateTime timestamp;

    public CurrencyConversionResponse() {
        this.timestamp = LocalDateTime.now();
    }

    // All getters and setters
    public String getFromCurrency() {
        return fromCurrency;
    }

    public void setFromCurrency(String fromCurrency) {
        this.fromCurrency = fromCurrency;
    }

    public String getToCurrency() {
        return toCurrency;
    }

    public void setToCurrency(String toCurrency) {
        this.toCurrency = toCurrency;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public double getConvertedAmount() {
        return convertedAmount;
    }

    public void setConvertedAmount(double convertedAmount) {
        this.convertedAmount = convertedAmount;
    }

    public double getExchangeRate() {
        return exchangeRate;
    }

    public void setExchangeRate(double exchangeRate) {
        this.exchangeRate = exchangeRate;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }
}

File: frontend_mobile/myAlkansyaMob/app/src/main/java/com/example/myalkansyamobile/model/CurrencyConversionResponse.kt
package com.example.myalkansyamobile.model

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class CurrencyConversionResponse(
    @SerializedName("fromCurrency")
    val fromCurrency: String,
    
    @SerializedName("toCurrency")
    val toCurrency: String,
    
    @SerializedName("amount")
    val amount: Double,
    
    @SerializedName("convertedAmount")
    val convertedAmount: Double,
    
    @SerializedName("exchangeRate")
    val exchangeRate: Double,
    
    @SerializedName("timestamp")
    val timestamp: String
)

data class ConversionResult(
    @SerializedName("conversion")
    val conversion: CurrencyConversionResponse,
    
    @SerializedName("trends")
    val trends: Map<String, TrendData>
)

data class TrendData(
    @SerializedName("historicalRate")
    val historicalRate: Double,
    
    @SerializedName("changePercent")
    val changePercent: Double,
    
    @SerializedName("trendPoints")
    val trendPoints: List<Double>
)

